packages/fx-core/src/plugins/resource/bot/configs/teamsBotConfig.ts

```diff
@@ -23,60 +23,60 @@ export class TeamsBotConfig {
   public actRoles: PluginActRoles[] = [];
   public resourceNameSuffix = "";
 
-  public async restoreConfigFromContext(context: PluginContext): Promise<void> {
-    await this.scaffold.restoreConfigFromContext(context);
-    await this.provision.restoreConfigFromContext(context);
-    await this.localDebug.restoreConfigFromContext(context);
-    await this.deploy.restoreConfigFromContext(context);
+  private constructor() {
+    this.scaffold = new ScaffoldConfig();
+    this.provision = new ProvisionConfig();
+    this.localDebug = new LocalDebugConfig();
+    this.deploy = new DeployConfig();
+  }
+
+  public static async restoreConfigFromContext(context: PluginContext): Promise<TeamsBotConfig> {
+    let teamsBotConfig = new TeamsBotConfig();
```

Comment:
Nit: could be `const`. When we say `const` in JS, it means the reference is `const` instead of the object.

Reply:
ok

Reply:
BTW, if it reset config in every restoreConfigFromContext. Would it drop any  in-memory information that's shared among the command, e.g. if preprovision wants to pass some value to provision or postprovision. Please check your design.

Reply:
There are too may replace work in this PR. Ask Ivan to check the design first. I think even we want to reset these field, we don't need to involve such large change.

Reply:
@henzhang-ms there is also a step "saveToContext", will sync botconfig to context, and next step it will restore config from context.

Reply:
The design has a drawback, once we need to shared some internal variable, we needs to be write it into *.env.json and expose to user. It is fine to me, just for notice.